Post Request Advertisement

### Status:
Het post request zou moeten werken alleen krijg ik een error door LocalDate.

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: nl.avans.drivioapp, PID: 15270
    java.lang.IllegalArgumentException: Unable to create converter for java.util.List<nl.avans.drivioapp.model.Advertisement>
        for method DrivioApiService.getAdvertisements
        at retrofit2.Utils.methodError(Utils.java:54)
        at retrofit2.HttpServiceMethod.createResponseConverter(HttpServiceMethod.java:126)
        at retrofit2.HttpServiceMethod.parseAnnotations(HttpServiceMethod.java:85)
        at retrofit2.ServiceMethod.parseAnnotations(ServiceMethod.java:39)
        at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:202)
        at retrofit2.Retrofit$1.invoke(Retrofit.java:160)
        at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
        at $Proxy2.getAdvertisements(Unknown Source)
        at nl.avans.drivioapp.viewModel.AdvertisementViewModel$getAdvertisements$1.invokeSuspend(AdvertisementViewModel.kt:25)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:367)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
        at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
        at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
        at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
        at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
        at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
        at nl.avans.drivioapp.viewModel.AdvertisementViewModel.getAdvertisements(AdvertisementViewModel.kt:24)
        at nl.avans.drivioapp.viewModel.AdvertisementViewModel.<init>(AdvertisementViewModel.kt:20)
        at java.lang.Class.newInstance(Native Method)
        at androidx.lifecycle.ViewModelProvider$NewInstanceFactory.create(ViewModelProvider.kt:202)
        at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:322)
        at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:304)
        at androidx.lifecycle.ViewModelProvider$AndroidViewModelFactory.create(ViewModelProvider.kt:278)
        at androidx.lifecycle.SavedStateViewModelFactory.create(SavedStateViewModelFactory.kt:128)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:187)
        at androidx.lifecycle.ViewModelProvider.get(ViewModelProvider.kt:153)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:53)
        at androidx.lifecycle.ViewModelLazy.getValue(ViewModelLazy.kt:35)
        at nl.avans.drivioapp.DiscoverFragment.onViewCreated$lambda$0(DiscoverFragment.kt:29)
        at nl.avans.drivioapp.DiscoverFragment.onViewCreated(DiscoverFragment.kt:30)
        at androidx.fragment.app.Fragment.performViewCreated(Fragment.java:3128)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:552)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1899)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1817)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1760)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2985)
        at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2895)
        at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:263)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:351)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:248)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1543)
        at android.app.Activity.performStart(Activity.java:8330)
E/AndroidRuntime:     at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3670)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7872)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@fd18281, Dispatchers.Main.immediate]
    Caused by: java.lang.IllegalArgumentException: Platform class java.time.LocalDate requires explicit JsonAdapter to be registered
    for class java.time.LocalDate endDate
    for class nl.avans.drivioapp.model.Advertisement
    for java.util.List<nl.avans.drivioapp.model.Advertisement>
        at com.squareup.moshi.Moshi$LookupChain.exceptionWithLookupStack(Moshi.java:388)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:157)
        at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory.create(KotlinJsonAdapter.kt:262)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:145)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:105)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:74)
        at com.squareup.moshi.CollectionJsonAdapter.newArrayListAdapter(CollectionJsonAdapter.java:54)
        at com.squareup.moshi.CollectionJsonAdapter$1.create(CollectionJsonAdapter.java:38)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:145)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:105)
        at retrofit2.converter.moshi.MoshiConverterFactory.responseBodyConverter(MoshiConverterFactory.java:89)
        at retrofit2.Retrofit.nextResponseBodyConverter(Retrofit.java:362)
        at retrofit2.Retrofit.responseBodyConverter(Retrofit.java:345)
        at retrofit2.HttpServiceMethod.createResponseConverter(HttpServiceMethod.java:124)
        	... 57 more
     Caused by: java.lang.IllegalArgumentException: Platform class java.time.LocalDate requires explicit JsonAdapter to be registered
        at com.squareup.moshi.ClassJsonAdapter$1.create(ClassJsonAdapter.java:75)
        at com.squareup.moshi.Moshi.adapter(Moshi.java:145)
        	... 69 more
```

--- a/app/src/main/java/nl/avans/drivioapp/AdvertisementProfileFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/AdvertisementProfileFragment.kt
@@ -6,17 +6,17 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
 import nl.avans.drivioapp.databinding.FragmentAdvertisementProfileBinding
-import nl.avans.drivioapp.databinding.FragmentProfileBinding
+import nl.avans.drivioapp.databinding.FragmentDiscoverBinding
 
-class ProfileFragment : Fragment(R.layout.fragment_profile) {
-    private var _binding: FragmentProfileBinding? = null;
+class AdvertisementProfileFragment : Fragment(R.layout.fragment_advertisement_profile) {
+    private var _binding: FragmentAdvertisementProfileBinding? = null;
     private val binding get() = _binding!!;
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        _binding = FragmentProfileBinding.inflate(inflater, container, false);
+        _binding = FragmentAdvertisementProfileBinding.inflate(inflater, container, false);
         val view = binding.root;
         return view;
     }
@@ -24,8 +24,8 @@ class ProfileFragment : Fragment(R.layout.fragment_profile) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.btnAdvertisement.setOnClickListener {
-            replaceFragment(AdvertisementProfileFragment())
+        binding.btnCreateAdvertisement.setOnClickListener {
+            replaceFragment(CreateAdvertisementFragment())
         }
     }
 

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -77,11 +77,16 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
 
             // TODO: Navigate to something when the post request was successful
 
-//            advertisementViewModel.postAdvertisementResponse.observe(viewLifecycleOwner) {
-//                val response = advertisementViewModel.postAdvertisementResponse.value
-//                binding.tvResult.text = response?.body().toString()
-//            }
+            advertisementViewModel.postAdvertisementResponse.observe(viewLifecycleOwner) {
+                val response = advertisementViewModel.postAdvertisementResponse.value
 
+                if (response?.code() == 200) {
+                    Toast.makeText(activity, "Success!!", Toast.LENGTH_SHORT).show()
+                } else {
+                    Toast.makeText(activity, "Failed!!", Toast.LENGTH_SHORT).show()
+                }
+
+            }
         }
     }
 }

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -47,9 +47,9 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
         }
 
         dateRangePicker.addOnPositiveButtonClickListener {
-            val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
-            startDate = sdf.format(it.first)
-            endDate = sdf.format(it.second)
+            val dateFormatter = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+            startDate = dateFormatter.format(it.first)
+            endDate = dateFormatter.format(it.second)
 
             // TODO: Fix Resource string
             binding.tvStartDate.text = "Startdate: $startDate"

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -10,6 +10,7 @@ import androidx.fragment.app.viewModels
 import com.google.android.material.datepicker.MaterialDatePicker
 import nl.avans.drivioapp.databinding.FragmentCreateAdvertisementBinding
 import nl.avans.drivioapp.model.Advertisement
+import nl.avans.drivioapp.model.User
 import nl.avans.drivioapp.viewModel.AdvertisementViewModel
 import java.text.SimpleDateFormat
 import java.time.LocalDate
@@ -19,8 +20,8 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
     private var _binding: FragmentCreateAdvertisementBinding? = null;
     private val binding get() = _binding!!;
     private val advertisementViewModel: AdvertisementViewModel by viewModels()
-    private lateinit var startDate: LocalDate
-    private lateinit var endDate: LocalDate
+    private lateinit var startDate: String
+    private lateinit var endDate: String
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
@@ -44,11 +45,11 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
         }
 
         dateRangePicker.addOnPositiveButtonClickListener {
-            val sdf = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault())
-            val formattedStartDate = sdf.format(it.first)
-            val formattedEndDate = sdf.format(it.second)
-            startDate = LocalDate.parse(formattedStartDate)
-            endDate = LocalDate.parse(formattedEndDate)
+            val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+            startDate = sdf.format(it.first)
+            endDate = sdf.format(it.second)
+//            startDate = LocalDate.parse(formattedStartDate)
+//            endDate = LocalDate.parse(formattedEndDate)
 //            val startDate = sdf.format(it.first)
 //            val endDate = sdf.format(it.second)
 
@@ -71,11 +72,20 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
                 price,
                 startDate,
                 endDate,
-                29
+                User(29)
             )
 
             advertisementViewModel.postAdvertisement(advertisement)
+
+            // TODO: Navigate to something when the post request was successful
+//            replaceFragment(AdvertisementProfileFragment())
         }
+    }
+
+    private fun replaceFragment(fragment: Fragment) {
+        val fragmentTransaction = activity?.supportFragmentManager?.beginTransaction()
+        fragmentTransaction?.replace(R.id.flFragment, fragment)
+        fragmentTransaction?.commit()
     }
 
 }

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -8,6 +8,7 @@ import android.widget.Toast
 import androidx.core.view.get
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import androidx.lifecycle.LifecycleOwner
 import com.google.android.material.datepicker.MaterialDatePicker
 import nl.avans.drivioapp.databinding.FragmentCreateAdvertisementBinding
 import nl.avans.drivioapp.model.Advertisement
@@ -36,14 +37,9 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val advertisementViewModel: AdvertisementViewModel by viewModels()
-        advertisementViewModel.postAdvertisementResponse.observe(viewLifecycleOwner) {
-            Toast.makeText(context, "Advertisement created with response: ${advertisementViewModel.postAdvertisementResponse.value}", Toast.LENGTH_SHORT).show()
-        }
-
         val dateRangePicker =
             MaterialDatePicker.Builder.dateRangePicker()
-                .setTitleText("Select date")
+                .setTitleText("Select dates")
                 .build()
 
         binding.btnSelectDate.setOnClickListener {
@@ -54,10 +50,6 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
             val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
             startDate = sdf.format(it.first)
             endDate = sdf.format(it.second)
-//            startDate = LocalDate.parse(formattedStartDate)
-//            endDate = LocalDate.parse(formattedEndDate)
-//            val startDate = sdf.format(it.first)
-//            val endDate = sdf.format(it.second)
 
             // TODO: Fix Resource string
             binding.tvStartDate.text = "Startdate: $startDate"
@@ -80,19 +72,16 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
                 endDate,
                 User(29)
             )
-
             advertisementViewModel.postAdvertisementWithResponse(advertisement)
 
 
             // TODO: Navigate to something when the post request was successful
-//            replaceFragment(AdvertisementProfileFragment())
+
+//            advertisementViewModel.postAdvertisementResponse.observe(viewLifecycleOwner) {
+//                val response = advertisementViewModel.postAdvertisementResponse.value
+//                binding.tvResult.text = response?.body().toString()
+//            }
+
         }
     }
-
-    private fun replaceFragment(fragment: Fragment) {
-        val fragmentTransaction = activity?.supportFragmentManager?.beginTransaction()
-        fragmentTransaction?.replace(R.id.flFragment, fragment)
-        fragmentTransaction?.commit()
-    }
-
 }

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -1,0 +1,81 @@
+package nl.avans.drivioapp
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.get
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.google.android.material.datepicker.MaterialDatePicker
+import nl.avans.drivioapp.databinding.FragmentCreateAdvertisementBinding
+import nl.avans.drivioapp.model.Advertisement
+import nl.avans.drivioapp.viewModel.AdvertisementViewModel
+import java.text.SimpleDateFormat
+import java.time.LocalDate
+import java.util.*
+
+class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisement) {
+    private var _binding: FragmentCreateAdvertisementBinding? = null;
+    private val binding get() = _binding!!;
+    private val advertisementViewModel: AdvertisementViewModel by viewModels()
+    private lateinit var startDate: LocalDate
+    private lateinit var endDate: LocalDate
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentCreateAdvertisementBinding.inflate(inflater, container, false);
+        val view = binding.root;
+        return view;
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val dateRangePicker =
+            MaterialDatePicker.Builder.dateRangePicker()
+                .setTitleText("Select date")
+                .build()
+
+        binding.btnSelectDate.setOnClickListener {
+            activity?.let { it1 -> dateRangePicker.show(it1.supportFragmentManager, "DATE_PICKER") }
+        }
+
+        dateRangePicker.addOnPositiveButtonClickListener {
+            val sdf = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault())
+            val formattedStartDate = sdf.format(it.first)
+            val formattedEndDate = sdf.format(it.second)
+            startDate = LocalDate.parse(formattedStartDate)
+            endDate = LocalDate.parse(formattedEndDate)
+//            val startDate = sdf.format(it.first)
+//            val endDate = sdf.format(it.second)
+
+            // TODO: Fix Resource string
+            binding.tvStartDate.text = "Startdate: $startDate"
+            binding.tvEndDate.text = "Enddate: $endDate"
+        }
+
+        binding.btnConfirm.setOnClickListener {
+            val title = binding.etTitle.text.toString()
+            val description = binding.etDescription.text.toString()
+            val price = binding.etPrice.text.toString().toDouble()
+            val startDate = startDate
+            val endDate = endDate
+
+            val advertisement = Advertisement(
+                null,
+                title,
+                description,
+                price,
+                startDate,
+                endDate,
+                29
+            )
+
+            advertisementViewModel.postAdvertisement(advertisement)
+        }
+    }
+
+}

--- a/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
+++ b/app/src/main/java/nl/avans/drivioapp/CreateAdvertisementFragment.kt
@@ -4,6 +4,7 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.core.view.get
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
@@ -34,6 +35,11 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        val advertisementViewModel: AdvertisementViewModel by viewModels()
+        advertisementViewModel.postAdvertisementResponse.observe(viewLifecycleOwner) {
+            Toast.makeText(context, "Advertisement created with response: ${advertisementViewModel.postAdvertisementResponse.value}", Toast.LENGTH_SHORT).show()
+        }
 
         val dateRangePicker =
             MaterialDatePicker.Builder.dateRangePicker()
@@ -75,7 +81,8 @@ class CreateAdvertisementFragment : Fragment(R.layout.fragment_create_advertisem
                 User(29)
             )
 
-            advertisementViewModel.postAdvertisement(advertisement)
+            advertisementViewModel.postAdvertisementWithResponse(advertisement)
+
 
             // TODO: Navigate to something when the post request was successful
 //            replaceFragment(AdvertisementProfileFragment())

--- a/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
+++ b/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
@@ -47,7 +47,6 @@ class AdvertisementAdapter(
     }
 
     override fun onBindViewHolder(holder: AdvertisementViewHolder, position: Int) {
-        val sdf = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault())
         val item = dataset[position]
         holder.tvTitle.text = item.title
         holder.tvDescription.text = item.description

--- a/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
+++ b/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
@@ -9,6 +9,8 @@ import nl.avans.drivioapp.DiscoverFragment
 import nl.avans.drivioapp.R
 import nl.avans.drivioapp.databinding.ListAdvertisementBinding
 import nl.avans.drivioapp.model.Advertisement
+import java.text.SimpleDateFormat
+import java.util.*
 
 class AdvertisementAdapter(private val context: DiscoverFragment, private val dataset: List<Advertisement>) : RecyclerView.Adapter<AdvertisementAdapter.AdvertisementViewHolder>() {
 
@@ -28,12 +30,13 @@ class AdvertisementAdapter(private val context: DiscoverFragment, private val da
     }
 
     override fun onBindViewHolder(holder: AdvertisementViewHolder, position: Int) {
+        val sdf = SimpleDateFormat("yyyy/MM/dd", Locale.getDefault())
         val item = dataset[position]
         holder.tvTitle.text = item.title
         holder.tvDescription.text = item.description
         holder.tvPrice.text = item.price.toString()
-        holder.tvStartDate.text = item.startDate
-        holder.tvEndDate.text = item.endDate
+        holder.tvStartDate.text = sdf.format(item.startDate)
+        holder.tvEndDate.text = sdf.format(item.endDate)
 
     }
 

--- a/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
+++ b/app/src/main/java/nl/avans/drivioapp/adapter/AdvertisementAdapter.kt
@@ -35,8 +35,8 @@ class AdvertisementAdapter(private val context: DiscoverFragment, private val da
         holder.tvTitle.text = item.title
         holder.tvDescription.text = item.description
         holder.tvPrice.text = item.price.toString()
-        holder.tvStartDate.text = sdf.format(item.startDate)
-        holder.tvEndDate.text = sdf.format(item.endDate)
+        holder.tvStartDate.text = item.startDate
+        holder.tvEndDate.text = item.endDate
 
     }
 

--- a/app/src/main/java/nl/avans/drivioapp/model/Advertisement.kt
+++ b/app/src/main/java/nl/avans/drivioapp/model/Advertisement.kt
@@ -2,11 +2,10 @@ package nl.avans.drivioapp.model
 
 import com.squareup.moshi.Json
 import java.time.LocalDate
-import java.util.*
 
 data class Advertisement(
     @Json(name = "advertisementId")
-    val advertisementId: Int,
+    val advertisementId: Int?,
     @Json(name = "title")
     val title: String,
     @Json(name = "description")
@@ -14,7 +13,9 @@ data class Advertisement(
     @Json(name = "price")
     val price: Double,
     @Json(name = "startDate")
-    val startDate: String,
+    val startDate: LocalDate,
     @Json(name = "endDate")
-    val endDate: String
+    val endDate: LocalDate,
+    @Json(name="userId")
+    val userId: Int?
 )

--- a/app/src/main/java/nl/avans/drivioapp/model/Advertisement.kt
+++ b/app/src/main/java/nl/avans/drivioapp/model/Advertisement.kt
@@ -1,5 +1,6 @@
 package nl.avans.drivioapp.model
 
+import androidx.annotation.Nullable
 import com.squareup.moshi.Json
 import java.time.LocalDate
 
@@ -13,9 +14,9 @@ data class Advertisement(
     @Json(name = "price")
     val price: Double,
     @Json(name = "startDate")
-    val startDate: LocalDate,
+    val startDate: String,
     @Json(name = "endDate")
-    val endDate: LocalDate,
-    @Json(name="userId")
-    val userId: Int?
+    val endDate: String,
+    @Json(name="user")
+    val user: User?
 )

--- a/app/src/main/java/nl/avans/drivioapp/model/User.kt
+++ b/app/src/main/java/nl/avans/drivioapp/model/User.kt
@@ -5,16 +5,16 @@ import com.squareup.moshi.Json
 data class User(
     @Json(name = "userId")
     val userId: Int?,
-    @Json(name = "firstName")
-    val firstName: String,
-    @Json(name = "lastName")
-    val lastName: String,
-    @Json(name = "city")
-    val city: String,
-    @Json(name = "phone")
-    val phone: Long,
-    @Json(name = "email")
-    val email: String,
-    @Json(name = "passwords")
-    val passwords: String
+//    @Json(name = "firstName")
+//    val firstName: String,
+//    @Json(name = "lastName")
+//    val lastName: String,
+//    @Json(name = "city")
+//    val city: String,
+//    @Json(name = "phone")
+//    val phone: Long,
+//    @Json(name = "email")
+//    val email: String,
+//    @Json(name = "passwords")
+//    val passwords: String
 )

--- a/app/src/main/java/nl/avans/drivioapp/model/User.kt
+++ b/app/src/main/java/nl/avans/drivioapp/model/User.kt
@@ -1,0 +1,20 @@
+package nl.avans.drivioapp.model
+
+import com.squareup.moshi.Json
+
+data class User(
+    @Json(name = "userId")
+    val userId: Int?,
+    @Json(name = "firstName")
+    val firstName: String,
+    @Json(name = "lastName")
+    val lastName: String,
+    @Json(name = "city")
+    val city: String,
+    @Json(name = "phone")
+    val phone: Long,
+    @Json(name = "email")
+    val email: String,
+    @Json(name = "passwords")
+    val passwords: String
+)

--- a/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
+++ b/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
@@ -24,7 +24,7 @@ interface DrivioApiService {
     ): Response<Advertisement>
 
     @POST(value = "advertisement")
-    suspend fun postAdvertisementWithResponse(@Body advertisement: Advertisement): Response<Advertisement>
+    suspend fun postAdvertisementWithResponse(@Body advertisement: Advertisement): Response<Unit>
 
 }
 

--- a/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
+++ b/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
@@ -6,11 +6,16 @@ import nl.avans.drivioapp.model.Advertisement
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
 import retrofit2.converter.scalars.ScalarsConverterFactory
+import retrofit2.http.Body
 import retrofit2.http.GET
+import retrofit2.http.POST
 
 interface DrivioApiService {
     @GET("advertisement")
     suspend fun getAdvertisements(): List<Advertisement>
+
+    @POST(value = "advertisement")
+    suspend fun postAdvertisement(@Body advertisement: Advertisement): Advertisement
 }
 
 private val BASE_URL = "https://drivio.nl/api/v1/"

--- a/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
+++ b/app/src/main/java/nl/avans/drivioapp/service/DrivioApiService.kt
@@ -3,6 +3,7 @@ package nl.avans.drivioapp.service
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import nl.avans.drivioapp.model.Advertisement
+import retrofit2.Response
 import retrofit2.Retrofit
 import retrofit2.converter.moshi.MoshiConverterFactory
 import retrofit2.converter.scalars.ScalarsConverterFactory
@@ -16,6 +17,9 @@ interface DrivioApiService {
 
     @POST(value = "advertisement")
     suspend fun postAdvertisement(@Body advertisement: Advertisement): Advertisement
+
+    @POST(value = "advertisement")
+    suspend fun postAdvertisementWithResponse(@Body advertisement: Advertisement): Response<Advertisement>
 }
 
 private val BASE_URL = "https://drivio.nl/api/v1/"

--- a/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
+++ b/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
@@ -21,8 +21,8 @@ class AdvertisementViewModel : ViewModel() {
         get() = _getAdvertisementByIdResponse
 
 
-    private val _postAdvertisementResponse: MutableLiveData<Response<Advertisement>> = MutableLiveData()
-    val postAdvertisementResponse: LiveData<Response<Advertisement>>
+    private val _postAdvertisementResponse: MutableLiveData<Response<Unit>> = MutableLiveData()
+    val postAdvertisementResponse: LiveData<Response<Unit>>
         get() = _postAdvertisementResponse
 
     init {

--- a/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
+++ b/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
@@ -8,6 +8,7 @@ import kotlinx.coroutines.launch
 import nl.avans.drivioapp.model.Advertisement
 import nl.avans.drivioapp.service.DrivioApi
 import org.json.JSONArray
+import retrofit2.Response
 
 private const val TAG = "AdvertisementViewModel"
 
@@ -16,8 +17,8 @@ class AdvertisementViewModel : ViewModel() {
     val advertisementResponse: LiveData<List<Advertisement>>
         get() = _advertisementResponse
 
-    private val _postAdvertisementResponse: MutableLiveData<String> = MutableLiveData()
-    val postAdvertisementResponse: LiveData<String>
+    private val _postAdvertisementResponse: MutableLiveData<Response<Advertisement>> = MutableLiveData()
+    val postAdvertisementResponse: LiveData<Response<Advertisement>>
         get() = _postAdvertisementResponse
 
     init {
@@ -30,17 +31,9 @@ class AdvertisementViewModel : ViewModel() {
         }
     }
 
-    fun postAdvertisement(advertisement: Advertisement) {
-        viewModelScope.launch {
-            DrivioApi.retrofitService.postAdvertisement(advertisement)
-//            _advertisementResponse.value = "postAdvertisement: ${advertisement} posted"
-        }
-    }
-
     fun postAdvertisementWithResponse(advertisement: Advertisement) {
         viewModelScope.launch {
-            val postResponse = DrivioApi.retrofitService.postAdvertisementWithResponse(advertisement)
-            _postAdvertisementResponse.value = "Response code: ${postResponse.code()}"
+            _postAdvertisementResponse.value = DrivioApi.retrofitService.postAdvertisementWithResponse(advertisement)
         }
     }
 }

--- a/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
+++ b/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
@@ -25,4 +25,11 @@ class AdvertisementViewModel : ViewModel() {
             _advertisementResponse.value = DrivioApi.retrofitService.getAdvertisements()
         }
     }
+
+    fun postAdvertisement(advertisement: Advertisement) {
+        viewModelScope.launch {
+            DrivioApi.retrofitService.postAdvertisement(advertisement)
+//            _advertisementResponse.value = "postAdvertisement: ${advertisement} posted"
+        }
+    }
 }

--- a/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
+++ b/app/src/main/java/nl/avans/drivioapp/viewModel/AdvertisementViewModel.kt
@@ -16,6 +16,10 @@ class AdvertisementViewModel : ViewModel() {
     val advertisementResponse: LiveData<List<Advertisement>>
         get() = _advertisementResponse
 
+    private val _postAdvertisementResponse: MutableLiveData<String> = MutableLiveData()
+    val postAdvertisementResponse: LiveData<String>
+        get() = _postAdvertisementResponse
+
     init {
         getAdvertisements()
     }
@@ -30,6 +34,13 @@ class AdvertisementViewModel : ViewModel() {
         viewModelScope.launch {
             DrivioApi.retrofitService.postAdvertisement(advertisement)
 //            _advertisementResponse.value = "postAdvertisement: ${advertisement} posted"
+        }
+    }
+
+    fun postAdvertisementWithResponse(advertisement: Advertisement) {
+        viewModelScope.launch {
+            val postResponse = DrivioApi.retrofitService.postAdvertisementWithResponse(advertisement)
+            _postAdvertisementResponse.value = "Response code: ${postResponse.code()}"
         }
     }
 }

--- a/app/src/main/res/drawable/add_fill0_wght400_grad0_opsz48.xml
+++ b/app/src/main/res/drawable/add_fill0_wght400_grad0_opsz48.xml
@@ -1,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="48dp"
+    android:height="48dp"
+    android:viewportWidth="48"
+    android:viewportHeight="48">
+  <path
+      android:fillColor="#FF000000"
+      android:pathData="M22.5,38V25.5H10v-3h12.5V10h3v12.5H38v3H25.5V38Z"/>
+</vector>

--- a/app/src/main/res/layout/fragment_advertisement_profile.xml
+++ b/app/src/main/res/layout/fragment_advertisement_profile.xml
@@ -1,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    tools:context=".AdvertisementProfileFragment">
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/btnCreateAdvertisement"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|end"
+        android:layout_margin="16dp"
+        app:srcCompat="@drawable/add_fill0_wght400_grad0_opsz48"
+        />
+
+</FrameLayout>

--- a/app/src/main/res/layout/fragment_create_advertisement.xml
+++ b/app/src/main/res/layout/fragment_create_advertisement.xml
@@ -76,7 +76,7 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_marginTop="16dp"
-            android:text="Select Date"
+            android:text="Select Dates"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toBottomOf="@+id/tilPrice" />
@@ -98,6 +98,15 @@
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toEndOf="@+id/tvStartDate"
             app:layout_constraintTop_toBottomOf="@+id/btnSelectDate" />
+
+        <TextView
+            android:id="@+id/tvResult"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tvStartDate" />
 
         <Button
             android:id="@+id/btnConfirm"

--- a/app/src/main/res/layout/fragment_create_advertisement.xml
+++ b/app/src/main/res/layout/fragment_create_advertisement.xml
@@ -1,0 +1,116 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".CreateAdvertisementFragment">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/tilTitle"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="16dp"
+            android:layout_marginEnd="16dp"
+            android:hint="Title"
+            app:layout_constraintBottom_toTopOf="@+id/tilDescription"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent">
+
+            <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/etTitle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content" />
+
+
+        </com.google.android.material.textfield.TextInputLayout>
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/tilDescription"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="16dp"
+            android:hint="Description"
+            app:layout_constraintBottom_toTopOf="@+id/tilPrice"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tilTitle">
+
+            <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/etDescription"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content" />
+
+
+        </com.google.android.material.textfield.TextInputLayout>
+
+        <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/tilPrice"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="16dp"
+            android:hint="Price"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tilDescription">
+
+            <com.google.android.material.textfield.TextInputEditText
+                android:id="@+id/etPrice"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:inputType="numberDecimal" />
+
+
+        </com.google.android.material.textfield.TextInputLayout>
+
+        <Button
+            android:id="@+id/btnSelectDate"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            android:text="Select Date"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tilPrice" />
+
+        <TextView
+            android:id="@+id/tvStartDate"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            app:layout_constraintEnd_toStartOf="@+id/tvEndDate"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/btnSelectDate" />
+
+        <TextView
+            android:id="@+id/tvEndDate"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@+id/tvStartDate"
+            app:layout_constraintTop_toBottomOf="@+id/btnSelectDate" />
+
+        <Button
+            android:id="@+id/btnConfirm"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="16dp"
+            android:text="Create Advertisement"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent" />
+
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+
+</FrameLayout>

--- a/app/src/main/res/layout/fragment_profile.xml
+++ b/app/src/main/res/layout/fragment_profile.xml
@@ -1,16 +1,25 @@
 <?xml version="1.0" encoding="utf-8"?>
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     tools:context=".ProfileFragment">
 
-    <!-- TODO: Update blank fragment layout -->
-    <TextView
-        android:id="@+id/tvProfileFragment"
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:text="Hello Profile Fragment"
-        android:textSize="26sp" />
+        android:layout_height="match_parent">
+
+        <Button
+            android:id="@+id/btnAdvertisement"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="My Advertisements"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
 
 </FrameLayout>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -1,3 +1,5 @@
 <resources>
     <string name="app_name">DrivioApp</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
 </resources>

